Mobile/ShopLens/Droid/Helpers/ProductNameHelper.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace ShopLens.Droid.Helpers
+{
+    public static class ProductNameHelper
+    {
+        public static string GetProductNameFromString(string products)
+        {
+            string[] array = products.Split(' ');
+            string productName = string.Empty;
+
+            for (int i = 0; i <= array.Length - 1; i++)
+            {
+                if (i > 0)
+                {
+                    if (i == 1)
+                    {
+                        productName += array[i];
+                    }
+                    else
+                    {
+                        productName += " " + array[i];
+                    }
+                }
+            }
```

Comment:
First impression: what in the world is this loop :astonished: 

Again, if this was unit tested, I'd go to the unit test and look what input it expects and what output it produces. (Not saying you should go write those tests right now)

Ok, so you're skipping first element in the list for some reason. Perhaps it was the "add" word or something (why is this method responsible for that?). Anyway, in that case I would write instead of 

`if (i > 0)
{
// blabla
}
`

Something like this:

`if (i == 0) break;`

How is this better? Firstly, it reads better - "if it's the first element, skip it".

Secondly, it avoids nesting "if"s. [Link](https://www.reddit.com/r/csharp/comments/33puzj/is_it_a_good_or_bad_practice_reducing_nested_if/)

To sum up, I think this method should be rewritten. Either the first word (the verb "add"/"delete"?) should not be passed to the method, or the amount of words to skip should be passed as a parameter. (I think first is better, as it is more Solid - single responsibility).

The `for` loop should be rewritten into a `foreach`, if possible. Use LINQ methods like `.Skip(1)`, if absolutely needed, as they are more readable. Actually, with this much logic, I think the whole method can be rewritten as:

```
productName = String.Join(array.Skip(1), " ")     
```

Reply:
Oh ok, so I shouldn't write unit tests, gotcha. While writing this method I thought I was writing something I'd have written back during the first semester. I guess that was a warning sign to not write this stuff...